### PR TITLE
Handle non object_reference table references (e.g. lateral flatten function)

### PR DESCRIPTION
**Pull request by NiallRees**
Resolves https://github.com/sqlfluff/sqlfluff/issues/635

### CODE REVIEW

File: test/core/rules/test_cases/L031.yml
Comment (dmateusp) [inline test/core/rules/test_cases/L031.yml:107]: +1 for issue number as the identifier

File: src/sqlfluff/core/rules/std.py
Comment (dmateusp) [inline src/sqlfluff/core/rules/std.py:3130]: that's clear :)

**Review by dmateusp — APPROVED** [review]

**Comment by NiallRees** [comment]
Thanks @dmateusp I'm going to hold off merging until the bit rule split pr gets merged.

**Pull request merged**
